### PR TITLE
Fixed description in Retry Policy topic

### PR DESCRIPTION
**Pull request by djmagee**
## What does this PR do?

Updates description and talks about Workflow Executions and Activity Executions in that order, per feedback from @dtornow.

### CODE REVIEW

**Review by flossypurse — APPROVED** [review]

File: docs/content/what-is-a-retry-policy.md [hunk 1]
```diff
@@ -5,9 +5,9 @@ tags:
   - explanation
 ---
 
-A Retry Policy is collection of attributes that instructs the Temporal Server how to retry a failure of an [Activity Task Execution](#activity-task-execution) or a [Workflow Execution](#workflow-execution).
+A Retry Policy is collection of attributes that instructs the Temporal Server how to retry a failure of a [Workflow Execution](#workflow-execution) or an [Activity Execution](#activity-execution).
```
Comment (flossypurse) [inline docs/content/what-is-a-retry-policy.md:8]: We will need to remember to fix these links at some point -

File: docs/content/what-is-a-retry-policy.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ tags:
   - explanation
 ---
```
Comment (flossypurse) [inline docs/content/what-is-a-retry-policy.md:6]: Let's add a "description" meta field here and copy that first sentence to it -

Comment (flossypurse) [reply docs/content/what-is-a-retry-policy.md:6]: rm the links in the description.

File: docs/content/what-is-a-retry-policy.md (hunk 1 above)
Comment (sw-yx) [reply docs/content/what-is-a-retry-policy.md:9]: its possible to leave TODOs in the frontmatter, wont matter at all

**Pull request merged**